eng/intellisense.targets

```diff
@@ -25,13 +27,26 @@
     <PackageDownload Include="Microsoft.Private.Intellisense" Version="[$(MicrosoftPrivateIntellisenseVersion)]" />
   </ItemGroup>
 
+  <!-- Prepare the intellisense package xml file by copying it to the project's intermediate folder and update its file timestamp.
+       This is necessary so that all project outputs are newer than all project inputs. Directly copying from the intellisense package
+       would violate that and break fast up-to-date check. -->
+  <Target Name="PrepareIntellisensePackageXmlFile"
+          Inputs="$(IntellisensePackageXmlFilePath)"
+          Outputs="$(IntermediateDocFileItemFromIntellisensePackage)">
+    <Copy SourceFiles="$(IntellisensePackageXmlFilePath)"
+          DestinationFiles="$(IntermediateDocFileItemFromIntellisensePackage)" />
+
+    <Touch Files="$(IntermediateDocFileItemFromIntellisensePackage)" />
+  </Target>
+
   <!-- Replace the compiler generated xml file in the obj folder with the one that comes from the intellisense package. -->
   <Target Name="ChangeDocumentationFileForPackaging"
+          DependsOnTargets="PrepareIntellisensePackageXmlFile"
           BeforeTargets="CopyFilesToOutputDirectory;DocumentationProjectOutputGroup"
           Condition="'$(UseCompilerGeneratedDocXmlFile)' != 'true' and '$(IntellisensePackageXmlFilePath)' != ''">
     <ItemGroup>
       <DocFileItem Remove="@(DocFileItem)" />
-      <DocFileItem Include="$(IntellisensePackageXmlFilePath)" />
+      <DocFileItem Include="$(IntermediateDocFileItemFromIntellisensePackage)" />
```

Comment:
Do we need to revert the name to the original? I wonder if the new suffix `*.intellisense-package*` would affect our process. It was my understanding that the intellisense file needs to match the assembly name, except for the extension.

Reply:
This is just the name of the intermediate file in the obj directory. We need a different name as we also have the compiler generated xml file. The final output path is the expected one: AssemblyName.xml.